.github/workflows/check.yml

```diff
@@ -1,43 +1,37 @@
-name: Check
+name: Pull Request Workflow
 
-# XXX disabled for now because we use Jenkins still, but at the time this was tested it was functional
-# on:
-#   pull_request:
-#     branches: [ master ]
+on:
+  pull_request:
+    branches:
+      - '*'  # Trigger on all branches for pull requests
 
 jobs:
-  check:
+  build:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
-
-    # Note: Actually we don't need Maven CLI, only Java 17, but this may have additional benefits due to the Maven cache
-    - name: Setup Maven
-      uses: s4u/setup-maven-action@v1.5.1
-      with:
-        java-version: 17
-        java-distribution: temurin
-        maven-version: 3.8.6
-
-    - name: Clean
-      run: ./build.sh clean
-      working-directory: ./build
-
-    - name: Test (commitStage)
-      run: ./build.sh commitStage
-      # run: ./build.sh integrationStage
-      working-directory: ./build
-
-      # https://github.com/marketplace/actions/junit-report-action
-    - name: Publish Test Report
-      uses: mikepenz/action-junit-report@v3
-      if: always() # always run even if the previous step fails
-      with:
-        report_paths: 'build/target/testReports/*.xml'
-
-      # TODO allure report?
-      # build/target/allure-results
-
-      # TODO archive logs?
-      # build/target/testReports/*.out,build/target/testReports/*.err
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup JAVA
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '17'
+
+      - name: Setup Maven
+        uses: s4u/maven-settings-action@v3.0.0
```

Comment:
Is there a specific reason you replaced the composite [setup-maven-action](https://github.com/marketplace/actions/setup-maven-action) with the individual ones?

You can see in the documentation that it is internally using 5 other actions.

Now the following is missing:

- Install/use a specific Maven version
- Maven specific cache that can reduce build times when there is a valid cache

The checkout was there before, but it was actually not needed as well.